jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ContextHandler.java

```diff
@@ -581,13 +590,38 @@ public void setLogger(Logger logger)
     @Override
     protected void doStart() throws Exception
     {
-        // If we are being started directly (rather than via a start of the CoreContextHandler), then
-        // we need to run ourselves in the core context
+        // If we are being started directly (rather than via a start of the CoreContextHandler),
+        // then we need the LifeCycle Listener to ensure both this and the CoreContextHandler are
+        // in STARTING state when doStartInContext is called.
         if (org.eclipse.jetty.server.handler.ContextHandler.getCurrentContext() != _coreContextHandler.getContext())
         {
+            // Make the CoreContextHandler lifecycle responsible for calling the doStartContext() and doStopContext().
```

Comment:
Just a thought... but would it be simpler to always have this handler as a non managed bean in the core context handler, and to always have a lifecycle listener registered that does the start and stop.   That way this is the normal path rather than a transient strange path?

Reply:
Note that if we did it this way, we'd need to explicitly dump the context handler from the core context handler, so that we would get the deep dump, but I think that is simple to do.

Reply:
@lachlan-roberts thoughts?

Reply:
Sorry didn't see this comment.

I think even if we had the listener permanently registered we would still need this `if` statement to manually start the `CoreContextHandler`.

Something like this
```java
if (org.eclipse.jetty.server.handler.ContextHandler.getCurrentContext() != _coreContextHandler.getContext())
     _coreContextHandler.start();
// else do nothing (don't call doStartInContext because listener will)
```

Reply:
And then would the listener `lifeCycleStarting` be called before or after this `doStart`? I'm not sure.

We would need to ensure this is in `STARTING` state when the listener is called (not just the `CoreContextHandler` in STARTING state).